common/djangoapps/util/keyword_substitution.py

```diff
@@ -57,9 +58,9 @@ def substitute_keywords(string, user=None, course=None):
         # Cannot proceed without course and user information
         return string
 
-    for key, func in KEYWORD_FUNCTION_MAP.iteritems():
+    for key, value in KEYWORD_FUNCTION_MAP.iteritems():
         if key in string:
-            substitutor = func(user, course)
+            substitutor = value[0](user, course)
```

Comment:
Could this be a named tuple or maybe even a dict? That way we can do something like:

```python
substitutor = value.substitute(user, course)
# or
substitutor = value['substitute'](user, course)
# the use of 'substitute above is an example; you could choose a more apt name
```

I get a bit uneasy about directly indexing into tuples like `value[0]`.